packages/wonder-blocks-icon/components/icon.test.js

```diff
@@ -0,0 +1,275 @@
+// @flow
+import * as React from "react";
+import * as Core from "@khanacademy/wonder-blocks-core";
+import icons from "../util/icon-assets.js";
+import {getPathForIcon, viewportPixelsForSize} from "../util/icon-util.js";
+
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
+
+// We mock things out so that we're in control of what really gets rendered.
+// Means we can test that we're using addStyle to generate the component
+// and then verify what that component has going on.
+jest.mock("@khanacademy/wonder-blocks-core", () => {
+    const mockStyledSVGComponent = jest.fn((props: any) => <div>Pretend</div>);
+    return {
+        _mockStyledSVGComponent: mockStyledSVGComponent,
+        addStyle: jest.fn(() => mockStyledSVGComponent),
+    };
+});
+
+// We cannot make the mock outside of the jest.mock call, so we make it inside
+// then attach it to the exports and grab it here.
+const mockStyledSVGComponent: JestMockFn<any, any> = (Core: any)
+    ._mockStyledSVGComponent;
+
+// Also, let's type up a couple of other mocks to shut flow up.
+const mockGetPathForIcon: JestMockFn<any, any> = (getPathForIcon: any);
+const mockViewportPixelsForSize: JestMockFn<
+    any,
+    any,
+> = (viewportPixelsForSize: any);
+
+jest.mock("../util/icon-util.js", () => ({
+    getPathForIcon: jest.fn(() => ({})),
+    viewportPixelsForSize: jest.fn(() => ({})),
+}));
+
+describe("Icon", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        unmountAll();
+    });
+
+    test("creates a styled svg using addStyle", async () => {
+        // Arrange
+        const importModulePromise = import("./icon.js");
+
+        // Act
+        await importModulePromise;
+
+        // Assert
+        expect(Core.addStyle).toHaveBeenCalledWith("svg");
+    });
+
+    test("applies aria-label to svg", async () => {
+        // Arrange
+        const {default: Icon} = await import("./icon.js");
+        const render = new Promise((resolve) => {
+            const nodes = (
+                <div>
+                    <Icon
+                        ref={() => resolve()}
+                        aria-label="ARIA LABEL"
+                        icon={icons.add}
+                    />
+                </div>
+            );
+
+            mount(nodes);
+        });
+
+        // Act
+        await render;
+
+        // Assert
+        expect(mockStyledSVGComponent).toHaveBeenCalledTimes(1);
+        expect(mockStyledSVGComponent.mock.calls[0][0]).toEqual(
+            expect.objectContaining({"aria-label": "ARIA LABEL"}),
```

Comment:
It seems like this test could be accomplished using a snapshot.  I know the snapshot tests are a bit noisy.  Once we get image comparison tests I'd like to eventually remove all visual CSS props from the snapshots so that we're only checking props like `aria-label` and other things that don't have any impact on the visuals.

Reply:
The snapshot tests test this with `addStyle`. These are unit tests, isolating the component from externals like `addStyle` to ensure the component is doing the right thing.